### ISSUE TITLE
Differences in the results between different runs without modifying the page

### ISSUE THREAD

**Issue by rawhihamad** [dequelabs/axe-core#3457]
Hello,

we are using axe-core npm library to do some scans on multiple web pages, there is one problem we are facing right now which is the tool shows a different number of violations and nodes each run, for violations it doesn't happen always but it happens each 5 to 10 runs, but the nodes happen every 1 to 3 runs, this makes it kind of hard to deal with from the developer's side.

the cases we faced up until now:

- new nodes appear on different pages where no changes happened on the pages (developer comment: if the violations are already there on the page then why didn't catch it from the first run?).
- in one run it finds a lot of nodes and in the next run, it finds none. (no changes happened on the pages)

we understand that the tool, for now, can catch an average of 57% of WCAG violations, but is this the cause of the issues we faced?
it will be great to understand the capabilities and the limitations of the tool so we can know how to proceed with our testing.

how we are using it:
we are using @axe-core/puppeteer npm package with puppeteer, we initialize to pages on different window sizes then we navigate to the page and wait for it to load using puppeteer like this `await this.pages[j].goto(url, {waitUntil: 'networkidle0', timeout: 60000 * 10});` then we analyze the page like this `results = await new AxePuppeteer(this.pages[j]).analyze();`, after that we get the results and do some work on them.

Our question is it expected from the tool to have these differences in the results at its current status or it's not an issue with it and there is something wrong we are doing? what we should expect from the tool?

Product: axe-core

Expectation: different runs should catch the same number of violations and nodes if there are no changes on the pages.

Actual: different runs catch a different number of nodes and sometimes violations as explained above.

Motivation: we need to have consistent results to move forward, we need to identify the problem so we can have a workaround or provide a solution.

<pre><code>
@axe-core/puppeteer version: 4.3.2
puppeteer version: 13.5.1
regenerator-runtime version: 0.13.9
@babel/core version: 7.16.12
@babel/preset-env version: 7.16.11
@babel/register version: 7.16.9

For Tooling issues:
- Node version: 17.7.1
- Platform:  Linux
</code></pre>

**Comment by WilcoFiers** [comment]
Axe-core should run the same way every time. There's nothing random about it. Differences in output come from differences in input. The only way I know in which axe-core can vary its output is if you run it against massive iframes, and you're bumping right up to the timeout limit of that. That would be very surprising. Far more likely is that there's some difference in the page you're not aware of. That's really common actually.

It can be subtle things, for example scroll position can change when content is loaded. Maybe you're starting axe-core a little too close to the onload event and occasionally your frontend library might not be done building the page. Ads, social media, third-party video, etc. might vary between page loads.

Generally I'd recommend not to focus on the numbers, but on the individual issues instead.

**Comment by rawhihamad** [comment]
@WilcoFiers many thanks for your reply, I'll discuss your points with the team and update the issue with any information.

**Comment by dylanb** [comment]
If you have a public page where you can produce an issue that you think is a bug, please share it with us and we can investigate

**Comment by rawhihamad** [comment]
@dylanb after a discussion with the team, we found a few heavy Iframes, some long page loading and a possibility of extra hand fixing some elements without being aware of the violations it might produce, we will address @WilcoFiers points and provide an update.
we talked also about a demo to reproduce the issue, we have some public pages but we need a period of 1 to 3 weeks to fix the issues from our side and have multiple runs then provide a full report to show a clear view of what elements we had the issue with.

Thanks to both of you for your support, I'll let you know once I have more updates.
